### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by hamza3202**

### CODE REVIEW

**Comment by stakater-user** [comment]
@hamza3202 Image is available for testing. `docker pull docker.io/stakater/scaler:SNAPSHOT-PR-1-2`

File: internal/pkg/providers/aws/aws.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	a.desiredCapacity = &scalerOptions.Desired
+	return nil
+}
+
+// Scale - Modifies auto scaling group.
+func (a *Aws) Scale() error {
+
+	// Initial credentials loaded from SDK's default credential chain. Such as
+	// the environment, shared credentials (~/.aws/credentials), or EC2 Instance
+	// Role. These credentials will be used to to make the STS Assume Role API.
```
Comment (rasheedamir) [inline internal/pkg/providers/aws/aws.go:44]: @hamza3202 it must support working with Role; and we should completely discourage using credentials; credentials can be used for local some sort of "local profile"

File: internal/pkg/cmd/common/common.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	Args    []string
+	Verbose bool
+}
+
+//ScalerOptions saves values of flags
+type ScalerOptions struct {
+	Options
+	ScalerName string
+	Provider   string
+	RoleArn    string
```
Comment (rasheedamir) [inline internal/pkg/cmd/common/common.go:26]: Is this generic and cross clouds?

Comment (hamza3202) [reply internal/pkg/cmd/common/common.go:26]: I am not familiar with other clouds and what might be required for them but the Provider value would be used to differentiate between different clouds in the future. and if they need a different strategy then role assumption then we would have to modify here.

File: internal/pkg/providers/aws/aws.go (hunk 1 above)
Comment (hamza3202) [reply internal/pkg/providers/aws/aws.go:44]: @rasheedamir when an ec2 instance is created, it is setup with role assigned via Launch Configuration. This reads that role. The role should ideally have the proper permissions to be able to assume the role passed in the arguments to Scaler.

**Comment by stakater-user** [comment]
@hamza3202 Image is available for testing. `docker pull docker.io/stakater/scaler:SNAPSHOT-PR-1-3`

**Comment by stakater-user** [comment]
@hamza3202 Image is available for testing. `docker pull docker.io/stakater/scaler:SNAPSHOT-PR-1-4`

**Comment by stakater-user** [comment]
@hamza3202 Image is available for testing. `docker pull docker.io/stakater/scaler:SNAPSHOT-PR-1-5`

**Comment by stakater-user** [comment]
@hamza3202 Image is available for testing. `docker pull docker.io/stakater/scaler:SNAPSHOT-PR-1-6`

**Review by kahootali — APPROVED** [review]

**Pull request merged**